### ISSUE TITLE
stage3d render bug

### ISSUE THREAD

**Issue by matrix3d** [Gamua/Adobe-Runtime-Support#262]
sometimes,not always.
![share](https://user-images.githubusercontent.com/4150631/76503741-5eaa4f80-6481-11ea-9ae5-87175b20bb5a.jpg)

**Comment by matrix3d** [comment]
iphone

**Comment by matrix3d** [comment]
![IMG_5101 PNG](https://user-images.githubusercontent.com/4150631/76504180-2b1bf500-6482-11ea-8183-2ed9003c6522.JPG)
![IMG_5100 PNG](https://user-images.githubusercontent.com/4150631/76504185-2ce5b880-6482-11ea-8114-31b96f13f5e1.JPG)

render bug,and no render bug,the same assets and code

**Comment by matrix3d** [comment]
[s10200.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4323241/s10200.zip)
the atf file

**Comment by matrix3d** [comment]
up

**Comment by mrchnk** [comment]
Hi @matrix3d,
I encountered the same problem a while ago.

- https://community.adobe.com/t5/flash-player/visual-artifacts-for-dxt-5-textures-black-blocks-in-pepperflash-plugin/td-p/9814357
- https://tracker.adobe.com/#/view/FP-4198484

Here I have a project, that can help reproduce that
https://mrchnk.github.io/FlashPlayerDXT5Bug/bin/index.html

I think the main source of the problem is JPEX-XR compression inside the ATF file. Seems like it cannot be decoded properly or deterministic for some reason. I may suggest you not to use that compression while creating textures (`-r` option of `png2atf`). 

To reduce file size you can create ATF with just block-compressed format, and after that compress it using LZMA or DEFLATE.

**Comment by matrix3d** [comment]
@mrchnk Is it possible to use as3 code to convert atf with -r to one without r

**Comment by mrchnk** [comment]
@matrix3d Yes. You can convert your textures from any proper ATF format back into bitmaps using AIR flash runtime with [Context3D.drawToBitmapData](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3D.html#drawToBitmapData())
